keps/sig-auth/1393-oidc-discovery/README.md

```diff
@@ -296,11 +303,208 @@ of versioning. However, we can still treat graduation in terms of
+      current dev cycle.
+
+* **Does enabling the feature change any default behavior?**
+  No. It adds an entirely new non-resource-url that can be used to discover
+  metadata related to the cluster's service account issuer.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+
+  No. This feature is always enabled post-GA.
+  The only way to roll back is to return to an older K8s version.
+
+  **Describe the consequences on existing workloads (e.g., if this is a runtime
+  feature, can it break the existing applications?).**
+
+  Existing applications would have to take a dependency on this feature to
+  be broken by it. Thus, enabling the feature for the first time is not a risk
+  to existing applications, but disabling it later could be.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+  The feature should continue to work just fine.
+
+* **Are there any tests for feature enablement/disablement?**
+
+  No.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Enablement shouldn't affect any existing workloads. If we broke the feature in
+  the future, we would _possibly_ see failures of workloads to authenticate to
+  Relying Parties _outside_ the cluster, but in-cluster workload to
+  kube-apiserver authentication would still work, since it doesn't rely
+  on this path.
+
+* **What specific metrics should inform a rollback?**
+  N/A
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  The standard upgrade tests would have covered this between alpha and beta,
+  when the feature was enabled by default.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+
+  No.
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Ideally, there would just be usage metrics for all API server endpoints.
+  Since we don't currently have that, the next best option would be to examine
+  API server logs.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [x] Other (treat as last resort)
+    - Details: API server logs, or ability of workloads to authenticate to
+      Relying Parties.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  We expect the endpoints to maintain high reliability, with reliability
+  matching that of kube-apiserver.
+
+* **Are there any missing metrics that would be useful to have to improve observability
+of this feature?**
+  It would be nice to have usage metrics for this endpoint. We haven't added
```

Comment:
Is there an existing issue to cover this? I am gathering that the issue is you are serving up a static resource and so any metrics plumbing would have to be in the basic HTTP server in the api server. You don't have any code hit after generating the static data, so you have nowhere to hook in a metric?

If that's the case, perhaps you can at least create an issue and get it on the radar of api machinery, if it's not already.

Reply:
Yeah more or less. Just registering a route doesn't seem to get you metrics by default. I could probably add some explicitly in the route handler, but a more transparent solution (with consistent route -> metric name mapping) would be preferable. 

Tracking issue: https://github.com/kubernetes/kubernetes/issues/98585